test/RequestHandlerTest.php

```diff
@@ -67,22 +30,59 @@ public function testRequestThrowsErrorOnMalformedBaseUrl()
 
     }
 
-    public function testRequestPost()
+    /**
+     * @expectedException Tumblr\API\RequestException
+     */
+    public function testRequestThrowsOnBadResponse()
     {
+        // Response stream
+        $stream = fopen('php://memory','r+');
+        fwrite($stream, '{"meta": {"status": 400, "msg": "Sadface."} }');
+        rewind($stream);
+
+        // Hook up response to all requests
+        $response = (new GuzzleHttp\Psr7\Response)
+            ->withStatus(400, 'Bad Request')
+            ->withBody(new GuzzleHttp\Psr7\Stream($stream));
+
+        $this->guzzle = $this->getMock('\GuzzleHttp\Client', array('request'));
+        $this->guzzle->expects($this->any())
+                     ->method('request')
+                     ->will($this->returnValue($response));
+
+        // Attached mocked guzzle client
         $client = new Tumblr\API\Client(API_KEY);
         $rh = $client->getRequestHandler();
-
         $rh->client = $this->guzzle;
 
-        $rh->setBaseUrl('/');
+        // Throws because it got a 400 back
+        $client->getBlogInfo('ceyko.tumblr.com');
+    }
+
+    public function testRequestGetsJsonResponseField()
+    {
+        // Response stream
+        $stream = fopen('php://memory','r+');
```

Comment:
instead of manually creating and writing to streams, might be able to utilize Guzzle HandlerStack to queue up responses. http://docs.guzzlephp.org/en/latest/testing.html